Documentation/op-guide/etcd-live-http-to-https-migration.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Enabling HTTPS in an existing etcd cluster
+---
+
+This guide outlines the process of migrating an existing etcd cluster from HTTP communication to encrypted HTTPS. For added security, it also shows how to require TLS peer certificates to authenticate connections.
+
+## Prepare cluster components
+
+### Check insecure port availability
+
+By default, etcd communicates with clients over two ports: 2379, the current and official IANA port designation, and 4001, for clients who may implement versions of the protocol older than 0.4. We leverage this quirk of legacy support to migrate a running cluster from insecure plain-HTTP communication (on the old port, 4001) to encrypted HTTPS (on the current port, 2379) without cluster downtime. We restrict the legacy communication on port 4001 to the local interface.
+
+If you've configured flannel, fleet, or other components to use custom ports, or 2379 only, they will be reconfigured to use port 4001.
```

Comment:
@vadimdemedes thanks for your first PR! There are couple of small build errors, please fix them to make the PR ready for the review.

Reply:
@spzala I've updated the document fixing "you" usage and the commit header format.
There are still three tests failing which doesn't seem to be related to my change: the first test fails because of the segmentation violation, the second test fails because of v3 the client integration test failure, and the third test seems to fail because of some wrong command line options.

Reply:
> @spzala I've updated the document fixing "you" usage and the commit header format.
> There are still three tests failing which doesn't seem to be related to my change: the first test fails because of the segmentation violation, the second test fails because of v3 the client integration test failure, and the third test seems to fail because of some wrong command line options.

@vikstrous thanks for addressing the comments and yes, the build failure is not related. A bigger event, KubeCon, is next week so review may take some time but we should have this reviewed soon.

Reply:
Hey @spzala, is there any update with reviewing this PR?

Reply:
Hello Valeriy, as you mentioned, this is same as an original content we had earlier. I would wait on @hexfusion to take a look. I am not sure if you looked at his comment in the issue, but he has suggested in the issue to `... to put this all together will full examples and link it from runtime-configuration docs [2] it could be useful` Thanks!